dspace-server-webapp/src/test/java/org/dspace/app/rest/BitstreamRestRepositoryIT.java

```diff
@@ -2279,6 +2291,223 @@ public void findByHandleAndFileNameForPublicItemWithEmbargoOnFile() throws Excep
+                                         .build();
+            bitstream3 = BitstreamBuilder.createBitstream(context, publicItem2, is)
+                                         .withName("Bitstream 3")
+                                         .withMimeType("text/plain")
+                                         .build();
+            bitstream4 = BitstreamBuilder.createBitstream(context, publicItem2, is)
+                                         .withName("Bitstream 4")
+                                         .withMimeType("text/plain")
+                                         .build();
+        }
+        context.restoreAuthSystemState();
 
+        // Add three out of four bitstreams to the list of bitstreams to be deleted
+        // But set the patch.operations.limit property to 2, so that the request is invalid
+        List<Operation> ops = new ArrayList<>();
+        RemoveOperation removeOp1 = new RemoveOperation(OPERATION_PATH_BITSTREAM_REMOVE + bitstream1.getID());
+        ops.add(removeOp1);
+        RemoveOperation removeOp2 = new RemoveOperation(OPERATION_PATH_BITSTREAM_REMOVE + bitstream2.getID());
+        ops.add(removeOp2);
+        RemoveOperation removeOp3 = new RemoveOperation(OPERATION_PATH_BITSTREAM_REMOVE + bitstream3.getID());
+        ops.add(removeOp3);
+        String patchBody = getPatchContent(ops);
+        String token = getAuthToken(admin.getEmail(), password);
+
+        Assert.assertTrue(bitstreamExists(token, bitstream1, bitstream2, bitstream3, bitstream4));
+        DSpaceServicesFactory.getInstance().getConfigurationService().setProperty("patch.operations.limit", 2);
+
+        getClient(token).perform(patch("/api/core/bitstreams")
+                                     .content(patchBody)
+                                     .contentType(MediaType.APPLICATION_JSON_PATCH_JSON))
+                        .andExpect(status().isBadRequest());
+
+        // Verify that no bitstreams were deleted since the request was invalid
+        Assert.assertTrue(bitstreamExists(token, bitstream1, bitstream2, bitstream3, bitstream4));
+    }
+
+    public boolean bitstreamExists(String token, Bitstream ...bitstreams) throws Exception {
+        for (Bitstream bitstream : bitstreams) {
+            if (getClient(token).perform(get("/api/core/bitstreams/" + bitstream.getID()))
+                                .andReturn().getResponse().getStatus() != SC_OK) {
+                return false;
+            }
+        }
+        return true;
+    }
+
+    public boolean bitstreamNotFound(String token, Bitstream ...bitstreams) throws Exception {
+        for (Bitstream bitstream : bitstreams) {
+            if (getClient(token).perform(get("/api/core/bitstreams/" + bitstream.getID()))
+                                .andReturn().getResponse().getStatus() != SC_NOT_FOUND) {
+                return false;
+            }
+        }
+        return true;
+    }
 
```

Comment:
tests with unpriviledged users and with authorized users that are not repository managers are missing